extensions/control-plane/callback/callback-static-endpoint/src/test/java/org/eclipse/edc/connector/callback/staticendpoint/CallbackStaticEndpointExtensionTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.callback.staticendpoint;
+
+import org.eclipse.edc.connector.spi.callback.CallbackRegistry;
+import org.eclipse.edc.junit.extensions.DependencyInjectionExtension;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.configuration.Config;
+import org.eclipse.edc.spi.system.configuration.ConfigFactory;
+import org.eclipse.edc.spi.system.injection.ObjectFactory;
+import org.eclipse.edc.spi.types.domain.callback.CallbackAddress;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.ArgumentCaptor;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.eclipse.edc.connector.callback.staticendpoint.CallbackStaticEndpointExtension.EDC_CALLBACK_SETTING_PREFIX;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(DependencyInjectionExtension.class)
+public class CallbackStaticEndpointExtensionTest {
+
+    CallbackStaticEndpointExtension extension;
+
+    CallbackRegistry callbackRegistry = mock(CallbackRegistry.class);
+
+    ServiceExtensionContext context;
+
+    @BeforeEach
+    void setUp(ServiceExtensionContext context, ObjectFactory factory) {
+        context.registerService(CallbackRegistry.class, callbackRegistry);
+        extension = factory.constructInstance(CallbackStaticEndpointExtension.class);
+        this.context = spy(context);
+    }
+
+    @Test
+    void initialize_shouldConfigureMultipleCallbacks() {
+
+        var cb1 = CallbackAddress.Builder.newInstance()
```

Comment:
seems a lot of code to test config parsing, why not put config as it should be, e.g. `ConfigFactory.fromMap(Map.of(`edc.callback....`))`
and then verifying that the callback registry has been called in a certain way?

Reply:
Yeah it's a left over, it was testing more stuff previously, i guess i can simplify this now